### PR TITLE
custom examples

### PR DESCRIPTION
**Pull request by ykethan**
#### Description of changes:

adds cdk examples for Geo, Analytics, REST API, In-App messaging
modifies the existing custom example to use addOutput method from backend

Tested each using Amplify JS v6  
related: https://github.com/aws-amplify/docs/pull/6963

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/directory/directory.mjs [hunk 1]
```diff
@@ -2103,6 +2103,18 @@ export const directory = {
                 },
                 {
                   path: 'src/pages/gen2/build-a-backend/add-aws-services/overriding-resources/index.mdx'
+                },
+                {
+                  path: 'src/pages/gen2/build-a-backend/add-aws-services/restapi/index.mdx'
```
Comment (josefaidt) [inline src/directory/directory.mjs:2108]: can this be `rest-api`?

File: src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export function getStaticProps(context) {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+Amplify enables you to collect analytics data for your App. The Analytics category uses Amazon Cognito Identity pools to identify users in your App. Cognito allows you to receive data from authenticated, and unauthenticated users in your App.
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx:14]: the capitalization of "App" stands out. Should this be just "app" to align with the quickstart language? 
image

Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx:14]:
```suggestion
Amplify enables you to collect analytics data for your app. The Analytics category uses Amazon Cognito Identity pools to _identify_ users in your app. Cognito allows you to receive data from authenticated and unauthenticated users in your app. 
```

File: src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+export function getStaticProps(context) {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+Amplify enables you to collect analytics data for your App. The Analytics category uses Amazon Cognito Identity pools to identify users in your App. Cognito allows you to receive data from authenticated, and unauthenticated users in your App. 
+We can use AWS CDK to create the necessary resources for analytics.
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx:15]:
```suggestion
We can use [AWS Cloud Development Kit (AWS CDK)](https://docs.aws.amazon.com/cdk/latest/guide/home.html) to create the necessary resources for analytics.
```

File: src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  auth, 
+  data,
+  // additional resources 
+});
+
+const analyticsStack = backend.createStack("analytics-stack");
+const pinpoint = new CfnApp(analyticsStack, "Pinpoint", {
+  name: "myPinpointApp",
+});
+
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx:31]:
```suggestion

// create an IAM policy to allow interacting with Pinpoint
```

File: src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+const pinpointPolicy = new Policy(analyticsStack, "PinpointPolicy", {
+  policyName: "PinpointPolicy",
+  statements: [
+    new PolicyStatement({
+      actions: ["mobiletargeting:UpdateEndpoint", "mobiletargeting:PutEvents"],
+      resources: [pinpoint.attrArn + "/*"],
+    }),
+  ],
+});
+
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx:41]:
```suggestion

// apply the policy to the authenticated and unauthenticated roles
```

File: src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(pinpointPolicy);
+backend.auth.resources.unauthenticatedUserIamRole.attachInlinePolicy(pinpointPolicy);
+
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/analytics/index.mdx:44]:
```suggestion

// patch the custom Pinpoint resource to the expected output configuration
```

**Review by josefaidt — CHANGES_REQUESTED** [review]

File: src/pages/gen2/build-a-backend/add-aws-services/rest-api/index.mdx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  };
+}
+
+Amplify provides APIs to invoke your endpoints from your web apps. The following is an example utilizing the [AWS Cloud Development Kit (AWS CDK)](https://docs.aws.amazon.com/cdk/latest/guide/home.html) to add a REST API resource powered by [API Gateway](https://aws.amazon.com/api-gateway/).
+
+Create a [Function](/gen2/build-a-backend/functions) with the following.
+
+```ts title="amplify/functions/my-function/handler.ts"
+import { APIGatewayProxyHandlerV2 } from "aws-lambda";
+export const handler: APIGatewayProxyHandlerV2 = async (event) => {
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/rest-api/index.mdx:20]:
```suggestion
import type { APIGatewayProxyHandlerV2 } from "aws-lambda";

export const handler: APIGatewayProxyHandlerV2 = async (event) => {
```

**Review by josefaidt — DISMISSED** [review]

**Review by josefaidt — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**